addons/mail/models/mail_message.py

```diff
@@ -1007,11 +1007,16 @@ def _message_format(self, fnames):
                     })
 
             if message_sudo.model and message_sudo.res_id:
-                record_name = self.env[message_sudo.model] \
+                record = self.env[message_sudo.model] \
                     .browse(message_sudo.res_id) \
                     .sudo() \
-                    .with_prefetch(thread_ids_by_model_name[message_sudo.model]) \
-                    .display_name
+                    .with_prefetch(thread_ids_by_model_name[message_sudo.model])
+                record_name = record.display_name
+
+                has_follower_access = self.env['mail.followers'].check_access_rights('read', raise_exception=False)
```

Comment:
I'm not expert on followers but isn't the access check on thread (`record`) enough?